rts/Game/Action.h

```diff
@@ -5,21 +5,35 @@
 
 #include <string>
 #include "Game/UI/KeySet.h"
-#include "Game/UI/KeySetSC.h"
 
 class Action
 {
 public:
 	Action() {}
 	Action(const std::string& line);
 
-	std::string command;   ///< first word, lowercase
-	std::string extra;     ///< everything but the first word
-	std::string rawline;   ///< includes the command, case preserved
-	std::string boundWith; ///< the string that defined the binding keyset
+	int         bindingIndex; ///< the order for the action trigger
+	std::string command;      ///< first word, lowercase
+	std::string extra;        ///< everything but the first word
+	std::string rawline;      ///< includes the command, case preserved
+	std::string boundWith;    ///< the string that defined the binding keyset
+	CKeyChain   keyChain;     ///< the bound keychain/keyset
 
-	CKeyChain keyChain;    ///< the bounded keychain/keyset
-	CKeyChainSC keyChainSC;
+	bool operator<(const Action& action) const
+	{
+		bool selfAnyMod = keyChain.back().AnyMod();
+		bool actionAnyMod = action.keyChain.back().AnyMod();
+
+		if (selfAnyMod == actionAnyMod)
+			return bindingIndex < action.bindingIndex;
+		else
+			return actionAnyMod;
```

Comment:
I'm not 100% convinced about this and whether it makes for a proper ordering. What is this trying to do?

Reply:
This pushes Any+ keysets to the end of the actionlist, this is done in current [here](https://github.com/beyond-all-reason/spring/blob/27d2b863c997111b9a5216c9dd7083b6b55c471c/rts/Game/UI/KeyBindings.cpp#L339-L344) by concatenating arrays.

Reply:
Basically the desired behavior is that Any+ keysets should have lower priority in the action trigger order, even if they are bound earlier.

I disagree with the idea, that seems justified only by the fact they didn't retain binding order as we do now, but I didn't want to change behavior.

Reply:
Ah fair, yeah then it works.